### PR TITLE
Win pkcs11 tests

### PR DESCRIPTION
**Pull request by dovholuknf**
Lost this comment when i made a new PR... Just porting the pkcs11 tests to work with both windows and linux

### CODE REVIEW

File: identity/engines/pkcs11/engine_windows_test.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	limitations under the License.
+*/
+
+package pkcs11
+
+import (
+	_ "net/url"
+)
+
+const pkcs11Lib = "V:/dev/tools/softhsm2/lib/softhsm2-x64.dll"
```
Comment (dovholuknf) [inline identity/engines/pkcs11/engine_windows_test.go]: ugh. need to convert this into an env var or something... any thoughts?

File: identity/engines/pkcs11/engine_linux_test.go [hunk 2]
```diff
@@ -16,9 +16,17 @@
...
-	_ "net/url"
-)
+import "os"
 
-const pkcs11Lib = "/usr/lib/softhsm/libsofthsm2.so"
+func getPkcs11Lib() (string) {
+	libFromEnv := os.Getenv(softHsmEnvVar)
+
+	if len(libFromEnv) > 0 {
+		log.Infof("LIB FROM ENV")
```
Comment (scareything) [inline identity/engines/pkcs11/engine_linux_test.go]: Do you intend to log the value of the env variable here? If not, this should probably be an `Info` vs `Infof`.
Also, I suggest logging the variable name so somebody doesn't need to look at the code to know what they need to set/change. e.g.:

    log.Infof("Using pkcs11 module %s, set via %s env var", libFromEnv, softHsmEnvVar)

File: identity/engines/pkcs11/engine_windows_test.go (hunk 1 above)
Comment (andrewpmartinez) [reply identity/engines/pkcs11/engine_windows_test.go]: If there is a default location use that as the const. Provide environment variable override. Ensure the logging tells someone which value is being used.

File: identity/engines/pkcs11/engine.go [hunk 3]
```diff
@@ -142,9 +142,10 @@ func (*engine) LoadKey(key *url.URL) (crypto.PrivateKey, error) {
 			return nil, fmt.Errorf("driver not specified for PKCS#11 engine, see docs")
 		}
 
-		driver = "lib" + key.Opaque + ".so"
+		driver = key.Opaque
 	}
 
+	log.Infof("using driver: %v", driver)
```
Comment (andrewpmartinez) [inline identity/engines/pkcs11/engine.go]: is this really info level?

File: identity/engines/pkcs11/engine_windows_test.go (hunk 1 above)
Comment (dovholuknf) [reply identity/engines/pkcs11/engine_windows_test.go]: make sure you have the latest PR to review. i addressed this

**Pull request merged**